File: DP/removing-dig.py
n = int(input())

INF = 10 ** 8

dp = [INF for _ in range(n+1)]

dp[0] = 0

for i in range(1, n+1):
    for j in str(i):
        dp[i] = min(1+dp[i-int(j)], dp[i])

print(dp[n])

File: DP/removing-dig.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int n;
    std::cin >> n;

    const int INF = 1e8;
    std::vector<int> dp(n + 1, INF);
    dp[0] = 0;

    for (int i = 1; i <= n; ++i) {
        for (char j : std::to_string(i)) {
            dp[i] = std::min(1 + dp[i - (j - '0')], dp[i]);
        }
    }

    std::cout << dp[n] << std::endl;

    return 0;
}
